File: src/services/criteriaService.js
import api from './api';

export const criteriaService = {
  getAll: () => api.get('/criteria'),
  create: (data) => api.post('/criteria', data),
  update: (id, data) => api.put(`/criteria/${id}`, data),
  delete: (id) => api.delete(`/criteria/${id}`),
};

File: backend/src/main/java/com/university/thesis/service/CriteriaService.java
package com.university.thesis.service;

import com.university.thesis.dto.request.CreateCriteriaRequest;
import com.university.thesis.dto.response.CriteriaResponse;

import java.util.List;

public interface CriteriaService {

    List<CriteriaResponse> getAll();
    CriteriaResponse create(CreateCriteriaRequest request);
    CriteriaResponse update(Long id, CreateCriteriaRequest request);
    void delete(Long id);
}
